torch/_inductor/codegen/triton.py

```diff
@@ -1862,9 +1877,15 @@ def codegen_kernel(self, name=None):
 
         code = IndentedBuffer()
 
-        size_hints = [
-            next_power_of_2(V.graph.sizevars.size_hint(numel)) for numel in self.numels
-        ]
+        size_hints = []
+        for numel in self.numels:
+            r = V.graph.sizevars.size_hint(numel)
+            if isinstance(r, sympy.Expr):
+                assert all(V.graph.sizevars.shape_env.is_unbacked_symint(s) for s in r.free_symbols)
+                # replace unbacked symints with default int value, for triton autotuning size hint purpose
+                r = sympy.expand(r.xreplace({s: sympy.Integer(32) for s in r.free_symbols}))
+            hint = next_power_of_2(r)
+            size_hints.append(hint)
```

Comment:
Mehh, I think I'm going to redo this bit, feeding a default int value is ok but I don't think we want to implement it the way you've done here (at minimum, size_hint shouldn't return symbolic stuff, as per @peterbell10 comment below)